### PR TITLE
Add missing routines and remove some issues in THM's LinearFP

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #28025

### CODE REVIEW

File: modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C [hunk 1]
```diff
@@ -84,7 +88,11 @@ LinearFluidProperties::c_from_v_e(Real v, Real e, Real & c, Real & dc_dv, Real &
   dc_de = 0;
 }
 
-Real LinearFluidProperties::cp_from_v_e(Real, Real) const { return 0; }
+Real
+LinearFluidProperties::cp_from_v_e(Real, Real) const
+{
+  return _cv;
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C:94]: Pr, mu, and k are user parameters (though they're optional...), so this should probably be computed as `k*Pr/mu`.

**Review by joshuahansel — CHANGES_REQUESTED** [review]
This class is probably pretty error-prone even after your changes, so we should first consider removing it - do we get anything useful from this class?

If we do want to keep it, one thing that's problematic is the zero default values for mu, k, Pr. Ideally, we'd throw an error if these values tried to be used when not provided, but I'm sure there are cases when they get called and then never used. So maybe defaulting to NaNs is more appropriate?

**Comment by GiudGiud** [comment]
> This class is probably pretty error-prone even after your changes, so we should first consider removing it - do we get anything useful from this class?

I would not remove it before it gets it first user haha
The interest came from a discussions post

**Comment by GiudGiud** [comment]
I ll address your review.

File: modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C (hunk 1 above)
Comment (joshuahansel) [reply modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C:103]: I see you deprecated `Pr`, so this is resolved 👍

**Comment by joshuahansel** [comment]
> I would not remove it before it gets it first user

That's the best time to remove it 😆 

The discussion in question is https://github.com/idaholab/moose/discussions/28010. Regardless, these fixes are fine to be applied. We can decide afterward whether to follow-up with deprecation of the class.

**Comment by GiudGiud** [comment]
> do we get anything useful from this class?

it's a nice example imo. It's almost simpler than SimpleFP and it has the potential to work with every single flow model once we figure out all the entropy routines

**Comment by GiudGiud** [comment]
Threaded failure is unrelated. app patch is merged. this should be good to go

File: modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C [hunk 2]
```diff
@@ -41,8 +44,14 @@ LinearFluidProperties::LinearFluidProperties(const InputParameters & parameters)
     _T_0(getParam<Real>("T_0")),
     _mu(getParam<Real>("mu")),
     _k(getParam<Real>("k")),
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C:46]:
```suggestion
    _k(getParam<Real>("k"))
```

File: modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C [hunk 3]
```diff
@@ -41,8 +44,14 @@ LinearFluidProperties::LinearFluidProperties(const InputParameters & parameters)
...
-    _Pr(getParam<Real>("Pr"))
+    _Pr(isParamSetByUser("Pr") ? getParam<Real>("Pr") : -1)
 {
+  if (_Pr == -1)
+    _Pr = _cv / _k * _mu;
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/src/fluidproperties/LinearFluidProperties.C:50]:
```suggestion
{
  if (isParamValid("Pr"))
    _Pr = getParam<Real>("Pr");
  else
    _Pr = _cv / _k * _mu;
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]

**Review by joshuahansel — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thank you for the review!

**Pull request merged**